src/v/storage/storage_resources.h

```diff
@@ -54,15 +54,15 @@ class adjustable_allowance : public ss::semaphore {
             return;
         }
 
-        auto reclaim = std::min(current(), _dirty);
+        auto reclaim = std::min(_sem.current(), _dirty);
         _dirty -= reclaim;
-        consume(reclaim);
+        _sem.consume(reclaim);
     }
 
     void set_capacity(uint64_t capacity) noexcept {
         _capacity = capacity;
         if (capacity > _capacity) {
-            signal(capacity - _capacity);
+            _sem.signal(capacity - _capacity);
         } else {
             _dirty += _capacity - capacity;
```

Comment:
can we just simply use `ss::consume_units` ? this way semaphore units will go negative, when all the held units will be released the counter will be updated accordingly. I think we wouldn't need `maybe_adjust` in this case ?

Reply:
I think you're right.

When I first wrote this, I didn't want to let the semaphore go negative on config changes, because some unit holders will not necessarily give up any units until the next time they take units and look at the checkpoint hint, so it would jam things up if the partitions holding units happened to be idle.  However, that isn't really a problem the way I've ended up using it: for the dirty_bytes allowances they aren't use in a blocking way, and for the inflight_* allowances the holders will give up units promptly.

I've simplified this to avoid the whole _dirty dance and use consume directly.